tests/drop_counters/test_drop_counters.py

```diff
@@ -320,21 +578,464 @@ def test_src_ip_is_loopback_addr(ptfadapter, duthost, setup, tx_dut_ports, pkt_f
+    """
+    ip_dst = None
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    vm_name = setup["mg_facts"]["minigraph_neighbors"][dut_iface]["name"]
+
+    for item in setup["mg_facts"]["minigraph_bgp"]:
+        if item["name"] == vm_name:
+            ip_dst = item["addr"]
+            break
+    if ip_dst is None:
+        pytest.skip("Testcase is not supported on current interface")
 
-    log_pkt_params(dut_iface, dst_mac, src_mac, "", pkt_fields["ip_src"])
+    log_pkt_params(dut_iface, dst_mac, src_mac, ip_dst, pkt_fields["ipv4_src"])
 
     pkt = testutils.simple_tcp_packet(
         eth_dst=dst_mac, # DUT port
         eth_src=src_mac, # PTF port
-        ip_src=pkt_fields["ip_src"], # PTF source
-        ip_dst="", # DUT source
+        ip_src=pkt_fields["ipv4_src"], # PTF source
+        ip_dst=ip_dst,
         tcp_sport=pkt_fields["tcp_sport"],
         tcp_dport=pkt_fields["tcp_dport"])
 
-    base_verification("L3", pkt, ptfadapter, duthost, setup["combined_drop_counter"], ptf_tx_port_id, tx_dut_ports[dut_iface])
+    base_verification("L3", pkt, ptfadapter, duthost, ptf_tx_port_id, tx_dut_ports[dut_iface])
+
+    # Verify packets were not egresed the DUT
+    exp_pkt = expected_packet_mask(pkt)
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+def test_ip_pkt_with_exceeded_mtu(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields, mtu_config):
+    """
+    @summary: Verify that IP packet with exceeded MTU is dropped and L3 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    tmp_port_mtu = 1500
+
+    log_pkt_params(dut_iface, dst_mac, src_mac, pkt_fields["ipv4_dst"], pkt_fields["ipv4_src"])
+    # Set temporal MTU. This will be restored by 'mtu' fixture
+    mtu_config.set_mtu(tmp_port_mtu, tx_dut_ports[dut_iface])
+
+    pkt = testutils.simple_tcp_packet(
+        pktlen=9100,
+        eth_dst=dst_mac, # DUT port
+        eth_src=src_mac, # PTF port
+        ip_src=pkt_fields["ipv4_src"], # PTF source
+        ip_dst=pkt_fields["ipv4_dst"], # VM IP address
+        tcp_sport=pkt_fields["tcp_sport"],
+        tcp_dport=pkt_fields["tcp_dport"]
+        )
+
+    send_packets(pkt, duthost, ptfadapter, ptf_tx_port_id)
+
+    # Verify L2 drop counter incremented on specific interface
+    l2_drops = get_pkt_drops(duthost, "portstat -j")[dut_iface][RX_ERR]
+    l2_drops = str_to_int(l2_drops)
+
+    if l2_drops != PKT_NUMBER:
+        fail_msg = "RX_ERR drop counter was not incremented on iface {}. DUT RX_ERR == {}; Sent pkts == {}".format(
+            dut_iface, l2_drops, PKT_NUMBER
+        )
+        pytest.fail(fail_msg)
+
+    # Skip L3 discards verification for platform with linked L2 and L3 drop counters
+    if not COMBINED_L2L3_DROP_COUNTER:
+        ensure_no_l2_drops(duthost)
```

Comment:
This seems like a repeat of the base_verification method.

Reply:
Same as above, totally missed RX_DRP vs. RX_ERR the first time through. I think it would be helpful to pass it as a parameter in case we have to go back and update the logic later - adding the parameter ensures that all the L2 tests will be consistent.

Reply:
Fixed.